### PR TITLE
Multicamera/esb 62

### PR DESCRIPTION
**Pull request by amsmv11**

### CODE REVIEW

File: brave/api/logic.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        return request, input
+
+    @staticmethod
+    def create_output(request, input):
+        output_uri = '/'.join(request.json['uri'].split("/")[:-2]) + "/live/" + request.json['uri'].split("/")[-1] + "0000" + str(input.id)
+        params = {'type': 'rtmp', 'uri': output_uri, 'source': 'input' + str(input.id)} 
+        request['session'].outputs.add(**params)
+        return request
+
+    def delete_input(input_content):
```
Comment (ArturFortunato) [inline brave/api/logic.py:20]: Marca estes como static methods

File: brave/api/route_handler.py [hunk 2]
```diff
@@ -48,8 +49,13 @@ async def get_health(request):
 
 
 async def delete_input(request, id):
-    input = _get_input(request, id)
-    run_on_master_thread_when_idle(input.delete)
+    Logic.delete_input(_get_input(request, id))
+
+    # input = _get_input(request, id)
```
Comment (ArturFortunato) [inline brave/api/route_handler.py:54]: já não precisamos disto

File: brave/api/route_handler.py [hunk 3]
```diff
@@ -110,17 +116,29 @@ async def update_mixer(request, id):
 
 
 async def create_input(request):
-    # TODO  adicionar aqui output com id correspondente ao input, mas com uri de rtmp://rmtp.eaglestream.tv/live/ID00001 00002 00003 etc... 
-
-    input = request['session'].inputs.add(**request.json)
-    input.setup()
-    logger.info('Created input #%d with details %s' % (input.id, request.json))
+    # TODO  adicionar aqui output com id correspondente ao input, mas com uri de rtmp://rmtp.eaglestream.tv/live/ID00001 00002 00003 etc...
```
Comment (ArturFortunato) [inline brave/api/route_handler.py]: Já não é TODO

File: brave/api/route_handler.py [hunk 4]
```diff
@@ -110,17 +116,29 @@ async def update_mixer(request, id):
...
+
+    # output_uri = '/'.join(request.json['uri'].split("/")[:-2]) + "/live/" + request.json['uri'].split("/")[-1] + "0000" + str(input.id)
+    # params = {'type': 'rtmp', 'uri': output_uri, 'source': 'input' + str(input.id)} 
+    # request['session'].outputs.add(**params)
+   
     return sanic.response.json({'id': input.id, 'uid': input.uid})
 
 
 async def create_output(request):
+    logger.info("route_handler.py")
```
Comment (ArturFortunato) [inline brave/api/route_handler.py]: Podemos tirar os loggers todos (ou quase todos?)

File: brave/inputoutputoverlay.py [hunk 5]
```diff
@@ -141,6 +141,9 @@ def summarise(self, for_config_file=False):
 
     @property
     def uid(self):
+        if hasattr(self,"name"):
+            return self.name
+        # if self.uid != self.input_output_overlay_or_mixer():
```
Comment (ArturFortunato) [inline brave/inputoutputoverlay.py]: Não precisamos disto

File: brave/inputoutputoverlay.py [hunk 6]
```diff
@@ -343,6 +346,7 @@ def _set_props(self, new_props, updating):
                         continue
 
                 if key == 'uri' and self.uid.startswith("input"):
+                    # self.logger.info()
```
Comment (ArturFortunato) [inline brave/inputoutputoverlay.py]: Não precisamos disto

**Review by ArturFortunato — CHANGES_REQUESTED** [review]

**Comment by FilipeReynaud** [comment]
Tó, from now on you should write your commit messages as "ESB-<#issue_number> bla bla" so that the commit ca be associated with the corresponding Jira issue. In this way, one can easily see your commits from Jira board and go directly to view your changes.

In your case: "ESB-62 added feature X".
This way, you can see something like the following (in Jira issues):

image

**Comment by ArturFortunato** [comment]
> Tó, from now on you should write your commit messages as "ESB-<#issue_number> bla bla" so that the commit ca be associated with the corresponding Jira issue. In this way, one can easily see your commits from Jira board and go directly to view your changes.
> 
> In your case: "ESB-62 added feature X".
> This way, you can see something like the following (in Jira issues):
> 
> image

I agree that he has to improve his commit messages, but the rules detect commits in the branch and ignore the messages, so that is not a blocker for the automation

**Comment by FilipeReynaud** [comment]
> > Tó, from now on you should write your commit messages as "ESB-<#issue_number> bla bla" so that the commit ca be associated with the corresponding Jira issue. In this way, one can easily see your commits from Jira board and go directly to view your changes.
> > In your case: "ESB-62 added feature X".
> > This way, you can see something like the following (in Jira issues):
> > image
> 
> I agree that he has to improve his commit messages, but the rules detect commits in the branch and ignore the messages, so that is not a blocker for the automation

Sorry to say this in this way, but you are wrong. The branch and commit messages only link to issue if they follow the nomenclature.
Check the issue corresponded to this pull request (there are no commit messages).

image
image

**Comment by ArturFortunato** [comment]
> > > Tó, from now on you should write your commit messages as "ESB-<#issue_number> bla bla" so that the commit ca be associated with the corresponding Jira issue. In this way, one can easily see your commits from Jira board and go directly to view your changes.
> > > In your case: "ESB-62 added feature X".
> > > This way, you can see something like the following (in Jira issues):
> > > image
> > 
> > 
> > I agree that he has to improve his commit messages, but the rules detect commits in the branch and ignore the messages, so that is not a blocker for the automation
> 
> Sorry to say this in this way, but you are wrong. The branch and commit messages only link to issue if they follow the nomenclature.
> Check the issue corresponded to this pull request (there are no commit messages).
> 
> image
> image

Funny boy, I created those rules. You do have to follow the nomenclature, but only on the branch name. You should follow that, but for the issue to move on the board, the rules only verify the state of the branch (nonexistent --> todo, existing --> in progress, on pull request --> to review, merged --> to QA). Now answer me on messenger

**Pull request merged**